### DISCUSSION TITLE
No ABI when publishing with --Move-2 flag

### DISCUSSION THREAD

**Discussion by magnum6actual** [aptos-labs/aptos-developer-discussions#435] [Questions]
### Describe your question in detail.

Publishing a module with CLI 4.1.0 and --move-2 flag will publish successfully without error.  But no ABI is published, it just says "N/A" in explorer.  Here's the module published with `aptos move publish`

https://explorer.aptoslabs.com/account/0x0c9297bd873d6f24422c2487ceac9657ab4ebb5813abf239f35a2633fd9b09e0/modules/code/simple?network=devnet

Here's the same module published with `aptos move publish --move-2`:

https://explorer.aptoslabs.com/account/0x5406f8b016667846da357434dd6ef932305b9d68461a57b625c0e26562cf9161/modules/code/simple?network=devnet

Am I missing a simple step here with --move-2 or is this a CLI issue?

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

Aptos CLI

### Describe your environment or tooling in detail

CLI 4.1.0

**Comment by magnum6actual** [comment]
I should add - publishing with --move-2 flag works fine with a localnet running from 4.1.0 and the ABI is generated. 

Perhaps the issue is simply that devnet is not move-2 ready yet?

**Comment by magnum6actual** [comment]
FYI - People starting to post about this issue.

https://x.com/0xAnto/status/1837185772371464448

**Comment by magnum6actual** [comment]
Answering my own question - this appears to be fixed with CLI 4.2.1

**Reply by gregnazario** [reply]
Interesting, is this because of the CLI upgrade, or because of the devnet upgrade?

When I looked at your links, they were already gone from devnet

**Comment by magnum6actual** [comment]
Well, as usual, @gregnazario is correct. It was the devnet update.  I just published the same code to f6c611c893e1e2257d008eb0c856f2ddd08054c137742d5708ca6a5c7520ec00 on devnet and testnet.  Devnet has the ABI.  Testnet does not.

**Comment by wrwg** [comment]
Just a heads up about our analysis around this issue.

1. It turns out that ABI and general display of module info for Move 2 features doesn't work in Aptos node release 1.19 and earlier. 
2. It works from 1.20 onwards. 1.20 is shortly before hitting testnet (~9/25).
3. Nevertheless, all other enum functionality works from 1.18 onwards, it is just the ABI related functionality and what depends in the API from it which bails out.

Thanks again for reporting!

**Reply by neoul** [reply]
@wrwg, If so, when does it apply to mainnet?

**Comment by wrwg** [comment]
@neoul tentative date for 1.20 to hit mainnet is October 3rd.
